### DISCUSSION TITLE
Updating MAUI???

### DISCUSSION THREAD

**Discussion by marshallellis** [dotnet/maui#14864] [Q&A]
Hello all:

We are new to MAUI so apologies if I'm asking this in the wrong place...
We are having some issues with the Entry control and the Completed events.
We cannot get the ENTER key to fire the completed event!?!?
Now we searched on this issue and this is already logged and apparently fixed BUT we've updated VS 2022 to latest and it is still not working??? The issue referenced that is was fixed in "7.0.0-rc.2.6866" ... How do I get this fix?

https://github.com/dotnet/maui/labels/fixed-in-7.0.0-rc.2.6866

Thank you

Marshall

**Comment by davidortinau** [comment]
@marshallellis VS 17.5.x is the latest stable release and includes the most current release of .NET 7. This would include the fix you're referring to. I confirmed this running 7.0.84 on Android API 33.

![Screenshot_1682983030](https://user-images.githubusercontent.com/41873/235547718-0c1b5d8e-984a-469d-8b85-53b7af8d7df3.png)

**Reply by marshallellis** [reply]
Thanks ... so I think I'm making a rookie mistake??? My machine is running .NET 4.8 and I would need to upgrade this???
Do I need to manually update to .NET Framework 7 before I can utilize updates from VS upgrades???


Microsoft Visual Studio Community 2022
Version 17.5.5
VisualStudio.17.Release/17.5.5+33627.172
**Microsoft .NET Framework
Version 4.8.09032**

Installed Version: Community

Visual C++ 2022   00482-90000-00000-AA596
Microsoft Visual C++ 2022

ASP.NET and Web Tools   17.5.320.55692
ASP.NET and Web Tools

Azure App Service Tools v3.0.0   17.5.320.55692
Azure App Service Tools v3.0.0

Azure Functions and Web Jobs Tools   17.5.320.55692
Azure Functions and Web Jobs Tools

C# Tools   4.5.2-3.23171.7+d17f741546fad2786cbd6394d08619544e53a36d
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Common Azure Tools   1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

Dotfuscator Community Edition   6.5.0+76d4669002
PreEmptive Protection - Dotfuscator CE

Extensibility Message Bus   1.4.3 (main@2a4517a)
Provides common messaging-based MEF services for loosely coupled Visual Studio extension components communication and integration.

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

Mono Debugging for Visual Studio   17.5.9 (11975e6)
Support for debugging Mono processes with Visual Studio.

NuGet Package Manager   6.5.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit https://docs.nuget.org/

Razor (ASP.NET Core)   17.5.2.2316603+9f1b6856460af1e592d387ebef416eadddac453f
Provides languages services for ASP.NET Core Razor.

SQL Server Data Tools   17.2.40120.0
Microsoft SQL Server Data Tools

TypeScript Tools   17.0.20105.2003
TypeScript Tools for Microsoft Visual Studio

Visual Basic Tools   4.5.2-3.23171.7+d17f741546fad2786cbd6394d08619544e53a36d
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual F# Tools   17.5.0-beta.23053.5+794b7c259d9646a7eb685dad865aa27da7940a21
Microsoft Visual F# Tools

Visual Studio IntelliCode   2.2
AI-assisted development for Visual Studio.

VisualStudio.DeviceLog   1.0
Information about my package

VisualStudio.Mac   1.0
Mac Extension for Visual Studio

VSPackage Extension   1.0
VSPackage Visual Studio Extension Detailed Info

Xamarin   17.5.0.173 (d17-5@33e727c)
Visual Studio extension to enable development for Xamarin.iOS and Xamarin.Android.

Xamarin Designer   17.5.3.46 (remotes/origin/d17-5@e4dd80b2bb)
Visual Studio extension to enable Xamarin Designer tools in Visual Studio.

Xamarin Templates   17.5.41 (ba80d05)
Templates for building iOS, Android, and Windows apps with Xamarin and Xamarin.Forms.

Xamarin.Android SDK   13.2.0.0 (d17-5/797e2e1)
Xamarin.Android Reference Assemblies and MSBuild support.
    Mono: 6dd9def
    Java.Interop: xamarin/java.interop/main@149d70fe
    SQLite: xamarin/sqlite/3.40.0@fdc1e34
    Xamarin.Android Tools: xamarin/xamarin-android-tools/main@9f02d77


Xamarin.iOS and Xamarin.Mac SDK   16.2.0.5 (7738c90c9)
Xamarin.iOS and Xamarin.Mac Reference Assemblies and MSBuild support.

**Reply by Xyncgas** [reply]
Install the latest visual studio preview, it will do everything for you

**Reply by Xyncgas** [reply]
if you are running MAUI you are already on .NET 6 / 7

**Reply by marshallellis** [reply]
OK ... apologies guys was a little lost. I have the .NET SDK 7.0.203 loaded which should have this fix BUT this issue still seems to persist. It looks like the virtual keypad works BUT the ENTER keys on the hardware keyboards for Honeywell/Zebra mobile scanners seem not to work work??? Am I missing something??? Device = Android 11
![image](https://user-images.githubusercontent.com/14200622/235658534-ed666de4-450d-4c85-88fe-35fd6ea4237f.png)

**Comment by marshallellis** [comment]
... Also to add ... these devices usually are set to automatically pass an ENTER key after a barcode is scanned which is ALSO not working just like the hardware keypad!
